docs/library/ubluetooth.rst

```diff
@@ -493,6 +514,64 @@ device name from the device information service).
     peripheral initiating the MTU exchange. NimBLE works for both roles.
 
 
+L2CAP connection-oriented-channels (CoC)
+----------------------------------------
+
+    This feature allows for socket-like data exchange between two BLE devices.
+    Once the devices are connected via GAP, either device can listen for the
+    other to connect.
+
+    **Note:** This is currently only supported when using the NimBLE stack on
+    STM32 and Unix (not ESP32). Only one L2CAP CoC may be active at a given time
+    (i.e. you cannot connect while listening).
+
+.. method:: BLE.l2cap_coc_listen(psm, mtu, /)
+
+    Start listening for incoming L2CAP CoC requests on the specified *psm*.
+
+    The *mtu* specified will be the maximum payload size that this device can
+    transmit (which may be different to the MTU used in the other direction).
+
+    When a remote device initiates a connection, the ``_IRQ_L2CAP_COC_ACCEPT``
+    event will be raised, followed by the ``_IRQ_L2CAP_COC_CONNECT`` event. Both
+    events include a "channel handle", and the connect event includes the local
+    and remote MTU.
+
+.. method:: BLE.l2cap_coc_connect(conn_handle, psm, mtu, /)
+
+    Connect to a listening peer on the specified *psm*.
+
+    The *mtu* specified will be the maximum payload size that this device can
+    transmit (which may be different to the MTU used in the other direction).
+
+    On successful connection, the the ``_IRQ_L2CAP_COC_CONNECT`` event will be
+    raised, including a "channel handle" and the local and remote MTU.
```

Comment:
what happens on an unsuccessful connection, is that even possible?  can it be attempting to make the connection forever?

Reply:
Good point. I've made this raise the `L2CAP_DISCONNECT` event (to mirror the way gap_connect() behaves) and added a `status` to the `L2CAP_DISCONNECT` event.

This happens, for example, if the remote end is not listening on that PSM.